From 89148c0dbbcdb421d0b5a246e0f08094c04da238 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 04 Oct 2021 17:19:42 +0300
Subject: [PATCH] [SW-52828] boot_if: introduce reg_width define

Change the magic 32 number with the REG_WIDTH define.
This is for preparation for the new internal interface addition.

Change-Id: I1c10aaa793861ca2bcec235deef3ec5c1f7b6744
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/lib/common/boot_if.c b/lib/common/boot_if.c
index f969a16..fd570fc 100644
--- a/lib/common/boot_if.c
+++ b/lib/common/boot_if.c
@@ -12,6 +12,8 @@
 #include <preboot.h>
 #include <hl_boot_if.h>			/* requires stdint.h */
 
+#define REG_WIDTH	32
+
 struct cpu_boot_if {
 	u32 status;	/* boot progress status register offset */
 	u32 response;
@@ -98,19 +100,19 @@
 void cpu_boot_err_clrset(u64 mask, bool set)
 {
 	cpu_boot_reg_clrset(cpu_boot_if.err[0], (u32)mask, set);
-	cpu_boot_reg_clrset(cpu_boot_if.err[1], (u32)(mask >> 32), set);
+	cpu_boot_reg_clrset(cpu_boot_if.err[1], (u32)(mask >> REG_WIDTH), set);
 }
 
 bool cpu_boot_err_get(u64 mask)
 {
 	return !!((cpu_boot_if.err_val[0] & (u32)mask) |
-		  (cpu_boot_if.err_val[1] & (u32)(mask >> 32)));
+		  (cpu_boot_if.err_val[1] & (u32)(mask >> REG_WIDTH)));
 }
 
 void cpu_boot_dev_sts_clrset(u64 mask, bool set)
 {
 	cpu_boot_reg_clrset(cpu_boot_if.dev_sts[0], (u32)mask, set);
-	cpu_boot_reg_clrset(cpu_boot_if.dev_sts[1], (u32)(mask >> 32), set);
+	cpu_boot_reg_clrset(cpu_boot_if.dev_sts[1], (u32)(mask >> REG_WIDTH), set);
 }
 
 void cpu_boot_rsp_set(enum cpu_msg_status sts)
@@ -127,5 +129,5 @@
 {
 	u64 val = cpu_boot_if.dev_sts_val[1];
 
-	return val << 32 | cpu_boot_if.dev_sts_val[0];
+	return val << REG_WIDTH | cpu_boot_if.dev_sts_val[0];
 }
